python-libraries/nanover-openmm/src/nanover/openmm/imd.py

```diff
@@ -171,6 +172,9 @@ def report(self, simulation: app.Simulation, state: mm.State) -> None:
             else:
                 frame_data.particle_positions = positions
                 frame_data.user_energy = self._total_user_energy
+                sparse_indices, sparse_forces = get_sparse_forces(self._user_forces)
```

Comment:
to satisfy the type checker you need to check that `self._user_forces` is not `None` before trying to use it, but I suggest if it's valid to instead not allow `self._user_forces` to be `None` and initialise it as an empty array to begin with